### PR TITLE
add a matrix guide

### PR DESCRIPTION
**Pull request by ryzokuken**
/cc @tc39/chairs @littledan @jasonwilliams @mpcsh

### CODE REVIEW

File: matrix-guide.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    1. Log into your account
+
+        Check the documentation for your preferred client. For most clients including Element, there should be a link to do this on the landing page when you're not already logged into another account.
+
+    1. Join the official TC39 channels
+
+        You can click through the following links to join the various official TC39 channels in your preferred client. They all correspond to the similarly titled channels on Freenode.
+
+        1. [#tc39-general]
```
Comment (ctcpip) [inline matrix-guide.md:39]: consider adding descriptions of channels.  here are some from what I was sent in a welcome email:

#tc39-general:matrix.org:
 public conversation, open to community members

#tc39-delegates:matrix.org:
 publicly viewable conversation, participation restricted to delegates

#temporaldeadzone:matrix.org:
 off-topic conversations and bachkchanneling. Technically public, but intended for delegates

#tc39-inclusion:matrix.org:
 an informal working group with the goal of making TC39 a more inclusive committee in which to participate.

**Review by mpcsh — APPROVED** [review]
🥳

**Comment by bakkot** [comment]
Might be worth mentioning the "verified" thing also? I found that insistent popup pretty confusing.

**Comment by ryzokuken** [comment]
> Might be worth mentioning the "verified" thing also? I found that insistent popup pretty confusing.

@bakkot interesting. I never see this because I verify all my devices but is it something that happens even if you don't have multiple devices?

**Comment by smorimoto** [comment]
I have never seen it on one device and/or multiple devices after verifying.

**Comment by ryzokuken** [comment]
Yeah, it never pops up unless you have unverified devices. The rationale on their side is pretty solid too: if there's a device that you don't recognize, you should either verify that it's you or revoke access.

**Comment by bakkot** [comment]
I just didn't know what "verify" meant. I clicked "yes" and it asked me for a "security key", which, I own [one of those](https://www.yubico.com/store/#yubikey-5-series) but I never set it up with Element, so I was confused. And apparently what it actually wanted was a recovery passphrase of some sort, which I also cannot recall having set up.

**Comment by ryzokuken** [comment]
yeah, that's weird. It shouldn't have asked for a security key unless you'd set one previously. I'll try to bring it up with the Element folks.

**Review by Jack-Works — APPROVED** [review]

**Review by jasonwilliams — APPROVED** [review]

File: matrix-guide.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+[list of clients]: https://matrix.org/clients/
+
+[#tc39-general]: https://matrix.to/#/!wbACpffbfxANskIFZq:matrix.org?via=matrix.org&via=igalia.com&via=mozilla.org
+[#tc39-delegates]: https://matrix.to/#/!WgJwmjBNZEXhJnXHXw:matrix.org?via=matrix.org&via=igalia.com&via=mozilla.org
+[#temporaldeadzone]: https://matrix.to/#/!RKGOsXKqdKdyWOiTEA:matrix.org?via=matrix.org&via=igalia.com
+[#tc39-implementers]: https://matrix.to/#/!hmsRHUEXriRovkvcin:matrix.org?via=matrix.org&via=igalia.com&via=mozilla.org
+[#tc39-ecma402]: https://matrix.to/#/!hmsRHUEXriRovkvcin:matrix.org?via=matrix.org&via=igalia.com&via=mozilla.org
+[#tc39-beginners]: https://matrix.to/#/!OXhgybpQzCtnugpzuz:matrix.org?via=matrix.org&via=igalia.com
+[#tc39-inclusion]: https://matrix.to/#/!DgpygRnlCHLTRbahDa:matrix.org?via=matrix.org&via=igalia.com&via=mozilla.org
+[#tc39-website]: https://matrix.to/#/!hmsRHUEXriRovkvcin:matrix.org?via=matrix.org&via=igalia.com&via=mozilla.org
```
[the remark this answers is not in the document]
Comment (jasonwilliams) [reply matrix-guide.md:58]: I was wondering that too @ryzokuken

[the remark this answers is not in the document]
Comment (ljharb) [reply matrix-guide.md:58]: Seems like we’d actively not want igalia (or any member company) referenced here, regardless

Comment (mpcsh) [inline matrix-guide.md:58]:
```suggestion
[#tc39-general]: https://matrix.to/#/!wbACpffbfxANskIFZq:matrix.org
[#tc39-delegates]: https://matrix.to/#/!WgJwmjBNZEXhJnXHXw:matrix.org
[#temporaldeadzone]: https://matrix.to/#/!RKGOsXKqdKdyWOiTEA:matrix.org
[#tc39-implementers]: https://matrix.to/#/!hmsRHUEXriRovkvcin:matrix.org
[#tc39-ecma402]: https://matrix.to/#/!hmsRHUEXriRovkvcin:matrix.org
[#tc39-beginners]: https://matrix.to/#/!OXhgybpQzCtnugpzuz:matrix.org
[#tc39-inclusion]: https://matrix.to/#/!DgpygRnlCHLTRbahDa:matrix.org
[#tc39-website]: https://matrix.to/#/!hmsRHUEXriRovkvcin:matrix.org
```

File: matrix-guide.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        You can click through the following links to join the various official TC39 channels in your preferred client. They all correspond to the similarly titled channels on Freenode.
+
+        1. [#tc39-general]
+        1. [#tc39-delegates]
+        1. [#temporaldeadzone]
+        1. [#tc39-implementers]
+        1. [#tc39-ecma402]
+        1. [#tc39-beginners]
+        1. [#tc39-inclusion]
+        1. [#tc39-website]
```
Comment (mpcsh) [inline matrix-guide.md:46]:
```suggestion
        1. [#tc39-general] - public conversation, open to community members
        1. [#tc39-delegates] - publicly viewable conversation, participation restricted to delegates
        1. [#temporaldeadzone] - off-topic conversations and bachkchanneling. Technically public, but intended for delegates
        1. [#tc39-implementers] - space for discussions regarding maintenance and development of the JavaScript engines
        1. [#tc39-ecma402] - space for discussion regarding maintenance and development of the `Intl` spec
        1. [#tc39-beginners] - space for clarifying questions on terminology, process, history, etc
        1. [#tc39-inclusion] - an informal working group with the goal of making TC39 a more inclusive committee in which to participate
        1. [#tc39-website] - for development of [tc39.es](https://tc39.es)
```

File: matrix-guide.md (hunk 1 above)
Comment (ryzokuken) [reply matrix-guide.md:39]: sounds great, I'd be happy to add descriptions for all of them.

File: matrix-guide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryzokuken) [reply matrix-guide.md:58]: I'm not entirely sure why Element adds them, I suspect it has something to do with the fact that everyone in those rooms is from one of those three servers. Removing that part still pops open the same page though, so I'm not sure what changed. Let's remove it and I'll try to follow up with matrix folks to see what this part does.

**Comment by ryzokuken** [comment]
Thanks @mpcsh for both your commits, merged them both into the PR.

Can everyone take a final look before we can merge this?

File: matrix-guide.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        Check the documentation for your preferred client. For most clients including Element, there should be a link to do this on the landing page when you're not already logged into another account.
+
+    1. Join the official TC39 channels
+
+        You can click through the following links to join the various official TC39 channels in your preferred client. They all correspond to the similarly titled channels on Freenode.
+
+        1. [#tc39-general] - public conversation, open to community members
+        1. [#tc39-delegates] - publicly viewable conversation, participation restricted to delegates
+        1. [#temporaldeadzone] - off-topic conversations and bachkchanneling. Technically public, but intended for delegates
```
Comment (ryzokuken) [inline matrix-guide.md:41]:
```suggestion
        1. [#temporaldeadzone] - off-topic conversations and backchanneling. Technically public, but intended for delegates
```

Comment (ryzokuken) [reply matrix-guide.md:41]: @mpcsh is this what you meant here?

**Comment by bakkot** [comment]
Looks good to me. There's some details which are probably worth adding - the tc39 "space", maybe a basic intro for people coming from IRC for how to get a more comfortable experience, but I think those can be iterated on after landing.

**Comment by ryzokuken** [comment]
> the tc39 "space"

I avoided doing that now because it's still a beta feature and might end up confusing someone, but I don't particularly mind. I've been personally rooting for spaces so long!

File: matrix-guide.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    Like IRC and unlike many closed platforms, Matrix is an open protocol. Therefore, you can choose which client(s) you would prefer to use based on your preferred workflow.
+
+    If you don't care too much and just want something that "works", we recommend [Element] (web, desktop and mobile).
+
+    If you want to explore further or experiment with alternate clients, check out this [list of clients].
+
+1. Create an account
+
+    (Skip this step if you already have a matrix account).
```
Comment (federicobucchi) [inline matrix-guide.md:13]: *Matrix (capitalized)

Comment (mpcsh) [inline matrix-guide.md:13]:
```suggestion
    (Skip this step if you already have a Matrix account).
```

**Comment by mpcsh** [comment]
@bakkot:
- I think explaining the TC39 space should probably be a follow-on once we stabilize it. I believe it's currently a beta feature, so we have to make some kind of decision about whether to use it now or not.
- A "here's how to make it more like IRC" guide would be fantastic! I don't want to volunteer you for more work, but I feel like it should be written by you or someone with similar experience / desires. Or, if you'd like to give me a list of things, I can put them into prose and into a separate PR.

@ryzokuken I think this is more than ready to land, and in the interest of not dragging it out, I'm just going to press the button. Thanks for your work!

**Comment by bakkot** [comment]
> A "here's how to make it more like IRC" guide would be fantastic!

[Done.](https://github.com/tc39/how-we-work/pull/90)

**Pull request merged**